packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -82,11 +91,26 @@ def __init__(
 
     async def on_message(self, message: HumanChatMessage):
         """
-        Method which receives a human message and processes it via
-        `self.process_message()`, calling `self.handle_exc()` when an exception
-        is raised. This method is called by RootChatHandler when it routes a
-        human message to this chat handler.
+        Method which receives a human message, calls `self.get_llm_chain()`, and
+        processes the message via `self.process_message()`, calling
+        `self.handle_exc()` when an exception is raised. This method is called
+        by RootChatHandler when it routes a human message to this chat handler.
         """
+
+        # check whether the configured LLM can support a request at this time.
+        if self.uses_llm and BaseChatHandler._requests_count > 0:
+            lm_provider_klass = self.config_manager.lm_provider
+            lm_provider_params = self.config_manager.lm_provider_params
+            lm_provider = lm_provider_klass(**lm_provider_params)
+
+            if not lm_provider.allows_concurrency:
+                self.reply(
+                    "Sorry, the currently selected language model cannot process more than one request at a time. Please wait for me to reply before sending another question.",
```

Comment:
```suggestion
                    "The currently selected language model can process only one request at a time. Please wait for me to reply before sending another question.",
```

Deletes "Sorry", phrases in a more positive way

Reply:
Thanks! This in the latest revision.